frontend/src/metabase/visualizations/visualizations/ObjectDetail.jsx

```diff
@@ -160,7 +160,12 @@ export class ObjectDetail extends Component {
       if (value === null || value === undefined || value === "") {
         cellValue = <span className="text-light">{t`Empty`}</span>;
       } else if (isa(column.special_type, TYPE.SerializedJSON)) {
-        const formattedJson = JSON.stringify(JSON.parse(value), null, 2);
+        let formattedJson;
+        try {
+          formattedJson = JSON.stringify(JSON.parse(value), null, 2);
+        } catch (e) {
+          formattedJson = JSON.stringify(value, null, 2);
```

Comment:
Since we didn't parse `value` as JSON, I don't think we need to `stringify` it here. We'd end up with quotes around the malformed JSON in the display.

e.g. `JSON.stringify('{"malformedJSON": ')` -> `"{\"malformedJSON\": "` rather than `{\"malformedJSON\": `.

Just this is probably fine though I guess `formattedJson` is a bit of a misnomer if we went into the `catch`.

```suggestion
          formattedJson = value;
```

Reply:
Hmmm okay, I just copied what was happening in the next code path.

![image](https://user-images.githubusercontent.com/509837/97894714-737d7100-1cf8-11eb-9894-ed2fdb3f6f43.png)

I figured this made sense because we don't know what type `value` is, but we can be confident `JSON.stringify` will return a string.

(Between you and me, I didn't actually verify my changes in a browser because the process of getting everything set up for just this change was a bit daunting. I just verified via the test passing.)

Reply:
I hadn't check in browser either, but I just did and it worked.